### ISSUE TITLE
[Bug]: Some images have invalid encoded perceptual hash value `00000000ffffffff`

### ISSUE THREAD

**Issue by gavinfoley** [YannickAlex07/imghash-rs#8]
### Version

1.1.0

### Describe the Bug

Hi there!  When I pass these two images through `perceptual_hash()`, then `.encode()`, I get what seems to me to be an invalid value, `00000000ffffffff`.  Other images work, so it's not a universal issue, but I'm not sure what's different about these.

This is the code I was using:
```rs
use std::{env, path::Path, process::exit};

use imghash::perceptual_hash;

fn main() {
    let args: Vec<String> = env::args().skip(1).collect();

    if args.len() != 1 {
        eprintln!("Error: Expected exactly one argument (path to image file)");
        exit(1);
    }

    let image_path = Path::new(&args[0]);

    let hash = match perceptual_hash(image_path) {
        Ok(hash) => hash,
        Err(e) => {
            eprintln!("Error: {}", e);
            exit(1);
        }
    };

    println!("{}", hash.encode());
}
```

With v1.1.0 of `imghash` and rustc 1.77.1 (7cf61ebde 2024-03-27) (stable-x86_64-pc-windows-msvc).

![invalid1](https://github.com/YannickAlex07/imghash-rs/assets/6389719/b5eca681-37aa-40b4-bbe9-4b8a6a861aec)
![invalid2](https://github.com/YannickAlex07/imghash-rs/assets/6389719/8abc7b8b-1652-4dbf-8a15-0195849bcf5e)

### On which platform did you observe this bug?

Windows

**Comment by YannickAlex07** [comment]
Hey, thanks for creating the issue! I will look into it, already have a guess on why it is happening. Will get back to you once I debugged it!

**Comment by YannickAlex07** [comment]
So I think I found the issue. It looks like the dct is not correctly applied before computing the hash. I will fix this until EOD and see if this will solve the issue.

**Comment by YannickAlex07** [comment]
I fixed the issue and publish version 1.1.1 that includes the fix. I already tried it with the images that you provided and the hashes seem to be correct now.

Please try again now and let me know if we can close the issue or if there are further issues 😄
